pkg/apis/app/v1beta1/application.go

```diff
@@ -55,14 +57,30 @@ func (a *Application) ExpectedResources(rsrc interface{}, rsrclabels map[string]
 	return &resource.ObjectBag{}, nil
 }
 
+// Strip the version part of gv
+func stripVersion(gv string) string {
+	if gv == "" {
+		return gv
+	}
+
+	re := regexp.MustCompile(`^[vV][0-9].*`)
+	// If it beings with only version, (group is nil), return empty string which maps to core group
+	if re.MatchString(gv) {
+		return ""
+	}
+
+	return strings.Split(gv, "/")[0]
+}
+
 // GKVersions returns the gvks for the given gk
 func GKVersions(s *runtime.Scheme, mgk metav1.GroupKind) []schema.GroupVersionKind {
 	gvks := []schema.GroupVersionKind{}
-	gk := schema.GroupKind{Group: mgk.Group, Kind: mgk.Kind}
+	gk := schema.GroupKind{Group: stripVersion(mgk.Group), Kind: mgk.Kind}
 	for gvk := range s.AllKnownTypes() {
 		if gk != gvk.GroupKind() {
 			continue
 		}
+		log.Printf(" %s ", gvk.String())
```

Comment:
nit: What's with the leading and trailing spaces?

Reply:
removed.